friends/fahrenheit/__init__.py

```diff
@@ -0,0 +1,45 @@
+import check50
+import check50.c
+
+@check50.check()
+def exists():
+    """fahrenheit.c exists"""
+    check50.exists("fahrenheit.c")
+
+@check50.check(exists)
+def compiles():
+    """fahrenheit.c compiles"""
+    check50.c.compile("fahrenheit.c", lcs50=True)
+
+@check50.check(compiles)
+def test37():
+    """37 degrees Celsius yields 98.6 degrees Fahrenheit"""
+    check50.run("./fahrenheit 37").stdout(number(98.6), "98.6\n").exit(0)
+
+@check50.check(compiles)
+def test0():
+    """0 degrees Celsius yields 32.0 degrees Fahrenheit"""
+    check50.run("./fahrenheit 0").stdout(number(32.0), "32.0\n").exit(0)
+
+@check50.check(compiles)
+def test100point00():
+    """100.00 degrees Celsius yields 212.0 degrees Fahrenheit"""
+    check50.run("./fahrenheit 100.00").stdout(number(212.0), "212.0\n").exit(0)
+
+@check50.check(compiles)
+def testneg40():
+    """-40 degrees Celsius yields -40.0 degrees Fahrenheit"""
+    check50.run("./fahrenheit -40").stdout(number(-40.0), "-40.0\n").exit(0)
+
+@check50.check(compiles)
+def test_lack_of_arguments():
+    """handles lack of command line arguments"""
+    check50.run("./fahrenheit").exit(1)
+
+@check50.check(compiles)
+def test_too_many_arguments():
+    """handles too many command line arguments"""
+    check50.run("./fahrenheit 0 32").exit(1)
+
+def number(num):
+    return fr"(?<!\d){num}(?!\d)"
```

Comment:
Ooh, just realized that you're passing in floating point numbers which means the decimal point is going to end up in the regex and so e.g. `3200` will pass the check that checks for `32.0` since `.` matches any character.

Simplest fix is just to replace `.` in `num` with `\.` which can be done by replacing `{num}` here with `{num.replace('.', '\\.')}`